all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst

```diff
@@ -0,0 +1,232 @@
+
+The Vagrant box build process uses more disk space than is available by default on the provided cloud images, so we will resize the image using ``qemu-img``. For our image, ``jammy-server-cloudimg-amd64.img``, run:
+
+.. code:: bash
+
+   sudo apt install qemu-utils
+   qemu-img resize jammy-server-cloudimg-amd64.img +10G
+
+This will add 10GB to the virtual disk of our image and give us enough space to complete the Vagrant box build.
+
+Launch a VM with QEMU
+---------------------
+We will use `SeaBIOS <https://www.seabios.org/SeaBIOS>`_ to boot our image. SeaBIOS booting an image is essentially the easiest as it is the default firmware used by QEMU.
+
+The following command launches a cloud image with:
+
+- KVM acceleration
+- the local machine's CPUs
+- 2GB of memory
+- no graphics and serial output to the console
+- snapshot will make writes to a temporary file instead of the disk image itself. This ensures the base disk is not touched. If at some point, you want to persist the changes you've made on the disk, press ``C-a s``
+- virtio network device that redirects guest port 22 to host's port 2222
+- virtio cloud image
+- virtio seed image
+
+.. code:: bash
+
+   qemu-system-x86_64  \
+     -cpu host -machine type=q35,accel=kvm -m 2048 \
+     -nographic \
+     -snapshot \
+     -netdev id=net00,type=user,hostfwd=tcp::2222-:22 \
+     -device virtio-net-pci,netdev=net00 \
+     -drive if=virtio,format=qcow2,file=./jammy-server-cloudimg-amd64.img \
+     -drive if=virtio,format=raw,file=seed.img
+
+.. note::
+   To exit the QEMU shell, press ``C-a x``
+
+Configure the environment
+-------------------------
+All the commands can be run in an ssh session except ``lb build`` that **has to be run via the serial console**. 
+
+Create and configure the build directory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   mkdir -p /build/auto && cd /build
+   ln -s /usr/share/livecd-rootfs/live-build/auto/* auto/
+
+Configure the environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Write this file on the VM. Modify it with your build parameters. 
+
+- ``SUITE``: same as VM Ubuntu series name 
+- ``IMAGE_TARGETS``: ``vagrant``
+- ``NOW``: current date (``YYYYMMDD``)
+
+.. code:: bash
+
+   cat << EOF > /build/inputs.sh
+   #########################################
+   # usually you don't need to change those
+
+   export PROJECT=ubuntu-cpc
+   export IMAGEFORMAT=ext4
+
+   #########################################
+   # Those are the same parameters used with ubuntu-bartender or livefs
+
+   # those should be the same as the VM running the build
+   export SUITE=jammy
+   export ARCH=amd64
+
+   # Defines the binary hooks that will be run.
+   # eg. azure, ec2, qcow2, oracle, etc...
+   # it is a comma separated list of strings
+   export IMAGE_TARGETS=vagrant
+
+   # this is the serial that will be set in /etc/cloud/build.info
+   export NOW=20240409
```

Comment:
maybe NOW=$(date +%Y%m%d)

that way they don't have to change it every single date. but :shrug:

Reply:
Good idea making it more generic.